script.js

```diff
@@ -1,12 +1,125 @@
 //You can edit ALL of the code here
+const allEpisodes = getAllEpisodes().map(episode =>{
+  const formattedSeason = (episode.season.toString()).padStart(2, "0") 
+  const formattedNumber = (episode.number.toString()).padStart(2, "0")
+  const episodeCorrectFormatV2 = `S${formattedSeason}E${formattedNumber}`
+
+  return {
+    ...episode,
+    formattedNumber: episodeCorrectFormatV2
+  }
+});
+const rootEl = document.getElementById("container")
+const selectEl = document.getElementById("selectEl")
+const headerEl = document.getElementById("headerEl")
+const searchBar = document.getElementById("searchBar")
+const selector = document.getElementById("selector")
+const counter = document.getElementById("episodeCounter")
+
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  // makePageForEpisodes(allEpisodes);
+  createEpisodeCards(allEpisodes)
+  optionCreator(allEpisodes)
+  //restarting value of search input
+  searchBar.value = ""
+  //restarting Total episode text
+  counter.innerHTML = `Total episodes: ${allEpisodes.length}`
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+// episode option creator
+function optionCreator (listOfEpisodes) {
+  listOfEpisodes.forEach(episode => { 
+    const selectionOption = document.createElement("option")
+    selectionOption.innerHTML= `${episode.formattedNumber} - ${episode.name}`
+    selector.appendChild(selectionOption)
+  })
+  //add event listener to selector
+  selector.addEventListener("change",filterEpisodeBySelect)
+  function filterEpisodeBySelect() {
+    const filteredEpisodes = allEpisodes.filter(episode => {
+      return episode.name.includes(selector.value.substring(9))
```

Comment:
Have a think about how this mechanism works and why it causes the bug I pointed out. How else could we do this?

Reply:
Here is my suggestion (don't click the arrow if you don't want to see it):

<details>        
  <summary>Change line 41 to:
  </summary> return selector.value.includes(episode.formattedNumber)    
</details> 